### ISSUE TITLE
A wrong error occurs when using dict literal with `**` expansion

### ISSUE THREAD

**Issue by 515hikaru** [microsoft/pyright#184]
**Describe the bug**

A wrong error occurs when using dict literal and `**` expansion.

**To Reproduce**

I get `'d1' is not accessed pyright` in the code below

```python
def merge():
    d1 = {'foo': 1}
    d2 = {'bar': 2}
    d3 = {**d1, **d2}
    print(d3)
```

also `'d2' is not accessed pyright` is shown.

**Expected behavior**

Above errors are not shown, because `d1` and `d2` is accessed.

**Screenshots or Code**

<img width="416" alt="screenshot-pyright" src="https://user-images.githubusercontent.com/34483340/62752940-6d242900-baa4-11e9-91dd-3e716b700527.png">

**VS Code extension or command-line**

* VS Code 1.37.0
* Pyright 1.0.53

**Additional context**

If you write d3 definition as follows, this problem does not occur.

```python
d3 = dict(**d1, **d2)
```

**Comment by erictraut** [comment]
Thanks for the bug report. I've addressed the problem, and the fix will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.54, which I just published.
